test/terra/backends/qasm_simulator/qasm_noise.py

```diff
@@ -148,3 +150,26 @@ def test_kraus_gate_noise(self):
                 **self.BACKEND_OPTS).result()
             self.assertSuccess(result)
             self.compare_counts(result, [circuit], [target], delta=0.05 * shots)
+
+class QasmKrausNoiseOnQFTTests:
+    """QasmSimulator Kraus error noise model on QFT tests."""
+
+    SIMULATOR = QasmSimulator()
+    BACKEND_OPTS = {}
+
+    def test_kraus_gate_noise_on_QFT(self):
+        """Test Kraus noise on a QFT circuit"""
+
+        shots = 1000
+        noise_models = ref_kraus_noise.kraus_gate_error_noise_models_full()
+
+        for noise_model in noise_models:
+            circuit = QFT(3)
+            ref_target = qi.Statevector(circuit).data
+            circuit.save_statevector()
+            job = execute([circuit], self.SIMULATOR, shots=shots,
```

Comment:
The last two lines are done again and again for every noise model (which is currently in fact one model)

Reply:
I changed the structure of the test - see comment in the next box.